pennylane/compiler/compiler.py

```diff
@@ -65,6 +71,14 @@ def _refresh_compilers():
         module_name = entry.module.split(".")[0]
         AvailableCompilers.names_entrypoints[module_name][entry.name] = entry
 
+    # Check the minimum version of Catalyst if installed
+    if "catalyst" in AvailableCompilers.names_entrypoints:
+        catalyst_version = metadata.version("pennylane-catalyst")
+        if Version(re.sub(r"\.dev\d+", "", catalyst_version)) < PL_CATALYST_MIN_VERSION:
+            raise CompileError(
+                f"PennyLane-Catalyst {PL_CATALYST_MIN_VERSION} or greater is required, but installed {catalyst_version}"
```

Comment:
One downside I see with this approach is that the error is raised regardless of whether the user wants to use PennyLane with Catalyst. As long as Catalyst is installed, the error will be raised on PennyLane import 🤔 

Ideally, the error would only be raised if the user tries to actually use `qml.qjit`.

Reply:
I didn't see this as a downside tbh as there's a two-way dependency between PL and Catalyst and this would encourage users to install the up-to-date and compatible Catalyst on their system. But, I can move the logic to `qml.qjit` if you think this is something unnecessary @dime10

Reply:
Thanks for pointing that out @dime10. Brought this up at stand up and the consensus seems to be that it would be much better if the error was raised inside `qjit`.

Reply:
Fixed [b9bca5d](https://github.com/PennyLaneAI/pennylane/pull/4979/commits/b9bca5d33d91ee587025f7843a80e65a9de04505) Thanks :t-rex: